pages/posts/dynamicref-and-generics.md

```diff
@@ -0,0 +1,172 @@
+---
+title: "Using Dynamic References to Support Generic Types"
+date: 2022-08-31
+tags:
+  - dynamic reference
+type: Engineering
+cover: /img/posts/2022/dynamicref-and-generics/cover.webp
+authors:
+  - name: Greg Dennis
+    photo: /img/avatars/gregsdennis.webp
+    link: https://twitter.com/gregsdennis
+    byline: JSON Tooling Implementer, Specification & Tooling Architect @Postman
+excerpt: "A step in the right direction for modelling data with JSON Schema"
+---
+
+One of the most common questions we get is how to represent a concept from a strongly typed programming language in JSON Schema.  Class hierarchies, polymorphisms, generics, etc.  These ideas define strongly typed languages, and influence our data models.
+
+The topic in this post can apply to any programming paradigm where you have a defined data model and support for something like generic types.  This will likely be the first in a disjoint series on the relation between data modelling in programming languages and JSON Schema.
+
+Today, I want to cover the idea of generics or templates or some other label you may have heard.  To start, let's cover what I mean by "generic types."  This isn't intended as a lesson on them; I just want to ensure that we're all on the same page.
+
+## Generic Types
+
+By "generic types" I mean the feature in many programming languages to create a type that requires the additional information of one or more secondary types to be complete.  In object-oriented programming, generics can also apply to services as well as data models, but since we use JSON Schema to describe data models, it's we can be fairly certain that the generic types we care about for this use case are wrappers and containers.
```

Comment:
There's a minor typo here.

> it's we can be fairly certain that the generic types we care about for this use case are wrappers and containers.

I guess the intention is just without the word "it's".

Reply:
I saw that, meant to change it, then couldn't find it again.  Thanks.